Source/CharismaPlugNPlay/Public/GameMode/CharismaGameMode.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Charisma Entertainment Ltd
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "MetaDataFunctionBase.h"
+#include "PlaythroughEntities.h"
+#include "GameFramework/GameModeBase.h"
+#include "Playthrough/CharismaPlayerComponent.h"
+#include "CharismaGameMode.generated.h"
+
+USTRUCT(BlueprintType)
+struct CHARISMAPLUGNPLAY_API FCharismaConnectionParams
+{
+	GENERATED_USTRUCT_BODY()
+
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = "Charisma|GameMode")
+	int StoryId;
+
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = "Charisma|GameMode")
+	int StoryVersion;
+	
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = "Charisma|GameMode")
+	FString ApiKey;
+
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = "Charisma|GameMode")
+	FString StartGraphRefId;
+};
+
+/**
+ * 
+ */
+UCLASS()
+class CHARISMAPLUGNPLAY_API ACharismaGameMode : public AGameModeBase
```

Comment:
I had a look in VS and I don't think this needs to be a game mode. Collating all the entities shouldn't require you having gamemode level access at the very least.

Just for context, the GameMode type has large lower-level gameplay responsibility in the context of UE(singleton handling numerous things like networking, player handling, custom gamestates, etc.) and will likely be used by most clients in some way or another.

What this means is that, if a client were to install the PnP into their existing project, with their existing game mode they'd have to manually reparent or edit the PnP code directly to suit their needs - which is smth we want to avoid. 

We also gain very little out of the GameMode parent here. I wouldn't say, move this into the playthrough either, because that really shouldn't be responsible for obtaining assets.

Instead imo, it should be its own UObject called CharismaPlaythroughData(v likely needs a better name), that works similar to current gamemode implementation:

- that you can either automatically CollateAllEntities
or
-assign each field manually in blueprint (probably need setters funcs here that update the PlaythroughEntities object on call)